rules/AnnotationsToAttributes/Rector/Class_/CoversAnnotationWithValueToAttributeRector.php

```diff
@@ -304,6 +303,10 @@ private function removeMethodCoversAnnotations(ClassMethod $classMethod): bool
                 continue;
             }
 
+            if (str_contains($desiredTagValueNode->value->value, '::') && ! $this->reflectionProvider->hasClass(self::COVERS_METHOD_ATTRIBUTE)) {
+                continue;
+            }
```

Comment:
this one seems can cover it, I am thinking of moving `CoversTrait` and `CoversMethod` to phpunit 11 rule tho, but that will be need in separate PR